src/NavBar/NavBar.jsx

```diff
@@ -1,45 +1,164 @@
-import React, { useMemo, cloneElement } from 'react'
+import React, { useMemo, useState, useRef } from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
 import css from '@styled-system/css'
-
-import { Flex } from '@ivoryio/kogaio'
 import { NavItem } from './Item'
 
+const GRADIENT_SIZE = 24
+
 export const NavBar = ({ children }) => {
-  const content = useMemo(() => {
-    if (!children.length) {
-      return null
+  const [activeIndex, setActiveIndex] = useState(0)
+
+  const main = useRef(null)
+
+  const items = useMemo(
+    () =>
+      children
+        ? children.map((child, idx) => ({
+            key: `nav-item-${idx}`,
+            idx: idx,
+            component: child
+          }))
+        : [],
+    [children]
+  )
+
+  const _scrollToItem = idx => () => {
+    if (!children || !children.length || idx <= 0) {
+      return setActiveIndex(0)
+    }
+
+    if (idx >= children.length - 1) {
+      return setActiveIndex(children.length - 1)
     }
+    setActiveIndex(idx)
+  }
 
-    const len = children.length
-    const w = `${100 / len}%`
+  const itemsContainerPosition = useMemo(() => {
+    if (!children || !children.length) {
+      return GRADIENT_SIZE
+    }
 
-    return Array.from(children).map(child =>
-      cloneElement(child, { maxWidth: w })
-    )
-  }, [children])
+    const getInnerWidth = () => {
+      if (!main.current) {
+        return 0
+      }
 
-  return <Container>{content}</Container>
-}
+      const items = main.current.getElementsByClassName('nav-bar-item')
+      let innerWidth = 0
+      const elementWidths = []
+      for (let i = 0; i < items.length; i++) {
+        const rect = items[i].getBoundingClientRect()
+        innerWidth += rect.width
```

Comment:
Can we achieve the same value using `offsetWidth` or something similar?
`getBoundingClientRect` is very expensive. [Here is why](https://medium.com/@alexcambose/the-problem-with-getboundingclientrect-is-that-webkit-recalculates-the-layout-of-the-dom-every-cd349b04dcdd)

Reply:
I know, but it's only the reliable way in this case.
Anyway, it's on click and we have animation. More, there are at max 5 elements, so it's not so expensive.